3.evaluate_model/scripts/nbconverted/apply_tsne.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Apply tsne to each feature dataset
+# 
+# Input: Data representations
+# Output: tsne embeddings for plotting
+
+# In[1]:
+
+
+import pathlib
+import numpy as np
+import pandas as pd
+from sklearn.manifold import TSNE
+
+import sys
+
+sys.path.append("../utils")
+from split_utils import get_features_data
+
+
+# In[2]:
+
+
+np.random.seed(1234)
+
+
+# In[3]:
+
+
+output_file = pathlib.Path("evaluations", "tsne_embeddings.csv.gz")
+
+
+# In[4]:
+
+
+# load x (features) and y (labels) dataframes
+labeled_data_path = pathlib.Path("../0.download_data/data/labeled_data__ic.csv.gz")
+labeled_data = get_features_data(labeled_data_path).reset_index(drop=True)
+
+print(labeled_data.shape)
+labeled_data.head(3)
+
+
+# In[5]:
+
+
+metadata_columns = [
+    "Mitocheck_Phenotypic_Class",
+    "Cell_UUID",
+    "Location_Center_X",
+    "Location_Center_Y",
+    "Metadata_Plate",
+    "Metadata_Well",
+    "Metadata_Frame",
+    "Metadata_Site",
+    "Metadata_Plate_Map_Name",
+    "Metadata_DNA",
+    "Metadata_Gene",
+    "Metadata_Gene_Replicate",
+    "Metadata_Object_Outline",
+]
+
+feature_groups = [
+    "CP",
+    "DP",
+    "CP_DP"
+]
+
+
+# ## Apply tSNE
+# 
+# We test different perplexities ranging from 2 to 300.
+# 
+# From scikit-learn (https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html):
+# 
+# > The perplexity is related to the number of nearest neighbors that is used in other manifold learning algorithms. Larger datasets usually require a larger perplexity. Consider selecting a value between 5 and 50. Different values can result in significantly different results. The perplexity must be less than the number of samples.
+# 
+# We do not know what the appropriate value of perplexity is for our dataset, so we will test several
```

Comment:
```suggestion
# We do not know what the appropriate value of perplexity is for our dataset, so we will test several.
```

Reply:
will address! Thanks for the careful eye 👁️